### ISSUE TITLE
Hystrix is a latency and fault tolerance library

### ISSUE THREAD

**Issue by codeanit** [codeanit/til#499]
# Hystrix
Hystrix is a latency and fault tolerance library designed to isolate points of access to remote systems, services and 3rd party libraries, stop cascading failure and enable resilience in complex distributed systems where failure is inevitable. 

### What is Hystrix?
Hystrix is designed to do the following:
    Give protection from and control over latency and failure from dependencies accessed (typically over the network) via third-party client libraries.
    Stop cascading failures in a complex distributed system.
    Fail fast and rapidly recover.
    Fallback and gracefully degrade when possible.
    Enable near real-time monitoring, alerting, and operational control.

In simple terms, it isolates external dependencies so that one may not affect the other.

Hystrix provides 2 isolation strategies, thread and semaphore based isolation, but it’s left to one to decide which one works based on the requirements.

# Resource
- [ ] https://github.com/Netflix/Hystrix
- [ ] https://medium.com/bliblidotcom-techblog/latency-and-fault-tolerance-for-distributed-system-2c6d3213e439
- [ ]
